validator/client/src/main/java/tech/pegasys/teku/validator/client/KeyManager.java

```diff
@@ -92,8 +104,46 @@ public DeleteKeysResponse deleteValidators(final List<BLSPublicKey> validators)
    * @param slashingProtection a combined slashing protection payload
    * @return a list of 1 status per keystore that was attempted to be imported
    */
-  public List<ImportStatus> importValidators(
+  public List<PostKeyResult> importValidators(
       final List<String> keystores, final List<String> passwords, final String slashingProtection) {
-    throw new NotImplementedException("importValidators not implemented yet");
+
+    final List<PostKeyResult> postKeyResults = new ArrayList<>();
+
+    if (keystores.size() == passwords.size()) {
+      final Iterator<String> keystoreIterator = keystores.iterator();
+      final Iterator<String> passwordIterator = passwords.iterator();
+      while (keystoreIterator.hasNext() && passwordIterator.hasNext()) {
+        try {
+          final String password = passwordIterator.next();
+          final KeyStoreData keystore = getKeystoreDataObject(keystoreIterator.next());
+          final BLSKeyPair keyPair =
+              new BLSKeyPair(
+                  BLSSecretKey.fromBytes(Bytes32.wrap(KeyStore.decrypt(password, keystore))));
+          if (keystore.getPubkey().equals(keyPair.getPublicKey().toSSZBytes())) {
+            postKeyResults.add(executeImport());
+          } else {
+            postKeyResults.add(PostKeyResult.error("Incorrect public key."));
+          }
+        } catch (JsonProcessingException e) {
+          postKeyResults.add(PostKeyResult.error("Invalid keystore."));
+        } catch (KeyStoreValidationException e) {
+          postKeyResults.add(PostKeyResult.error("Invalid password."));
+        }
+      }
+    } else {
+      postKeyResults.add(PostKeyResult.error("Keystores and passwords quantity must be the same."));
+    }
+    return postKeyResults;
+  }
+
+  private PostKeyResult executeImport() {
+    throw new NotImplementedException("executeImport not implemented yet.");
+  }
+
+  private KeyStoreData getKeystoreDataObject(final String keystore) throws JsonProcessingException {
+    final KeyStoreData keyStoreData;
+    keyStoreData = jsonMapper.readValue(keystore, KeyStoreData.class);
+    keyStoreData.validate();
+    return keyStoreData;
```

Comment:
I don't think it's safe to use our `ObjectMapper` instance here - the signers library has its own `ObjectMapper` instance configured in specific ways to ensure correct parsing of keystores.  I'd say we need to raise a PR on https://github.com/ConsenSys/signers/ to add a `KeyStoreLoader.loadFromString` method much like it's existing `loadFromFile` method but that accepts a string.

We'll have to do a release of signers once it's merged so we can pull the change into Teku but I can help with that.